domain/schema/model/sql/0023-relation.sql

```diff
@@ -0,0 +1,139 @@
+-- applications, or a peer relation.
+CREATE TABLE relation (
+    uuid TEXT NOT NULL PRIMARY KEY,
+    life_id INT NOT NULL,
+    relation_id INT NOT NULL,
+    CONSTRAINT fk_relation_life
+    FOREIGN KEY (life_id)
+    REFERENCES life (id)
+);
+
+-- The relation_unit table links a relation to a specific unit.
+CREATE TABLE relation_unit (
+    uuid TEXT NOT NULL PRIMARY KEY,
+    relation_uuid TEXT NOT NULL,
+    unit_uuid TEXT NOT NULL,
+    in_scope BOOLEAN DEFAULT FALSE,
+    departing BOOLEAN DEFAULT FALSE,
+    CONSTRAINT fk_relation_unit_uuid
+    FOREIGN KEY (unit_uuid)
+    REFERENCES unit (uuid),
+    CONSTRAINT fk_relation_uuid
+    FOREIGN KEY (relation_uuid)
+    REFERENCES relation (uuid)
+);
+
+CREATE UNIQUE INDEX idx_relation_unit
+ON relation_unit (relation_uuid, unit_uuid);
+
+-- The relation_unit_setting holds key value pair settings
+-- for a relation at the unit level. Keys must be unique
+-- per unit.
+CREATE TABLE relation_unit_setting (
+    relation_unit_uuid TEXT NOT NULL,
+    "key" TEXT NOT NULL,
+    value TEXT,
+    CONSTRAINT chk_key_empty CHECK ("key" != ''),
+    CONSTRAINT fk_relation_unit_uuid
+    FOREIGN KEY (relation_unit_uuid)
+    REFERENCES relation_unit (uuid),
+    PRIMARY KEY (relation_unit_uuid, "key")
+);
+
+-- The relation_application_setting holds key value pair settings
+-- for a relation at the application level. Keys must be unique
+-- per application.
+CREATE TABLE relation_application_setting (
+    relation_endpoint_uuid TEXT NOT NULL,
+    "key" TEXT NOT NULL,
+    value TEXT,
+    CONSTRAINT chk_key_empty CHECK ("key" != ''),
+    CONSTRAINT fk_relation_endpoint_uuid
+    FOREIGN KEY (relation_endpoint_uuid)
+    REFERENCES relation_endpoint (uuid),
+    PRIMARY KEY (relation_endpoint_uuid, "key")
+);
+
+-- The relation_status maps a relation to its status
+ -- as defined in the relation_status_type table.
+ CREATE TABLE relation_status (
+     relation_uuid TEXT NOT NULL PRIMARY KEY,
+     relation_status_type_id TEXT NOT NULL,
+     suspended_reason TEXT,
+     updated_at TIMESTAMP NOT NULL,
+     CONSTRAINT fk_relation_uuid
+     FOREIGN KEY (relation_uuid)
+     REFERENCES relation (uuid),
+     CONSTRAINT fk_relation_status_type_id
+     FOREIGN KEY (relation_status_type_id)
+     REFERENCES relation_status_type (id)
+ );
+
+CREATE TABLE relation_status_type (
+    id TEXT NOT NULL PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_relation_status_type_name
+ON relation_status_type (name);
+
+INSERT INTO relation_status_type VALUES
+(0, 'joining'),
+(1, 'joined'),
+(2, 'broken'),
+(3, 'suspending'),
+(4, 'suspended');
+
+-- The relation_sequence table is used to keep track of the
+-- sequence number for relation IDs within a model. Each
+-- relation must have an relation ID.
+CREATE TABLE relation_sequence (
+    -- The sequence number will start at 0 for each model and will be
+    -- incremented.
+    sequence INT NOT NULL DEFAULT 0
```

Comment:
You'll only want one row, see what the read only model does to prevent this.

Reply:
thanks

Reply:
You probably want a trigger to ensure that it's only ever monotonically increasing. Happy to land that in a subsequent PR.